### PR TITLE
Add Ability to display system time in the HUD and menus

### PR DESCRIPTION
**Pull request by MoonPadUSer**
Adds this toggle:
Radiobutton
that enables showing the current time in the lower right corner of the screen:
Main Screen:
Homescreen
In-Game:
In-Game
In the code I add drawing the system time in the `drawhud(...)` and the `drawbackground(...)` functions so it'll be drawn inside the menus (`drawbackground`) and in-game (`drawhud`).
One might argue that the time is pretty small, and I agree, but that is a good thing in my opinion, that way it doesn't distract gameplay but you'll still be able to keep an eye on the time :D

### CODE REVIEW

File: config/menus/options.cfg [hunk 1]
```diff
@@ -421,6 +421,12 @@ options_statistics = [
         ui_strut 1
         ui_radiobutton2 "more verbose" showfps 3
     ]
+    ui_text "On-Screen System Time:"
```
Comment (TheAssassin) [inline config/menus/options.cfg:424]: Include *where* it will pop up, please.

File: src/engine/main.cpp [hunk 2]
```diff
@@ -911,6 +912,7 @@ int main(int argc, char **argv)
     #endif
     #endif
 
+    setlocale( LC_ALL, "" );
```
Comment (TheAssassin) [inline src/engine/main.cpp:915]: I am absolutely no fan of such code. It is very likely to interfere with our localization plans.

Why is this needed? It's not even documented...

File: src/game/game.h [hunk 3]
```diff
@@ -1683,6 +1683,8 @@ namespace hud
     extern int raceinventory(int x, int y, int s, float blend);
     extern int drawscore(int x, int y, int s, int m, float blend, int count);
     extern void cleanup();
+    extern char const* system_time;
+    extern bool show_system_time();
```
Comment (TheAssassin) [inline src/game/game.h:1687]: You should be able to get away without some global static variable. This one is particularly annoying because it's polluting _all_ files' namespace.

File: src/game/hud.cpp [hunk 4]
```diff
@@ -7,6 +7,7 @@ namespace hud
 {
     const int NUMSTATS = 11;
     int damageresidue = 0, hudwidth = 0, hudheight = 0, lastteam = 0, laststats = 0, prevstats[NUMSTATS] = {0}, curstats[NUMSTATS] = {0};
+    char const* system_time = nullptr;
```
Comment (TheAssassin) [inline src/game/hud.cpp:10]: See above. Should go away. _At least_ from the global namespace.

File: src/game/hud.cpp [hunk 5]
```diff
@@ -39,6 +40,7 @@ namespace hud
     VAR(IDF_PERSIST, showloadingversion, 0, 1, 1);
     VAR(IDF_PERSIST, showloadingurl, 0, 1, 1);
 
+    VAR(IDF_PERSIST, showsystemtime, 0, 0, 1);
```
Comment (TheAssassin) [inline src/game/hud.cpp:43]: Aren't HUD options prefixed `hud` or so?

File: src/game/hud.cpp [hunk 6]
```diff
@@ -698,6 +700,11 @@ namespace hud
         result(list.getbuf());
     });
 
+    bool show_system_time()
```
Comment (TheAssassin) [inline src/game/hud.cpp:703]: This getter is kind of pointless given the fact the variable is accessible globally already.

File: src/game/hud.cpp [hunk 7]
```diff
@@ -3156,6 +3163,12 @@ namespace hud
                     default: break;
                 }
             }
+            if (showsystemtime)
```
Comment (TheAssassin) [inline src/game/hud.cpp:3166]: Blank lines in between these blocks wouldn't hurt anyone.

File: src/game/hud.cpp [hunk 8]
```diff
@@ -3156,6 +3163,12 @@ namespace hud
...
+            {
+                pushfont("console");
+                cy[1] -= draw_textf("%s", cx[1], cy[1], 0, 0, 255, 255, 255, bf, TEXT_RIGHT_UP, -1, bs, 1, system_time);
+                popfont();
+            }
```
Comment (TheAssassin) [inline src/game/hud.cpp:3171]: Another blank line below this makes the reviewer happy.

File: config/menus/options.cfg (hunk 1 above)
Comment (MoonPadUSer) [reply config/menus/options.cfg:424]: That would be inconsistent looking at the rest of the interface, e.g. On-Screen FPS

File: src/engine/main.cpp (hunk 2 above)
Comment (MoonPadUSer) [reply src/engine/main.cpp:915]: to be completely honest, I am not sure if it is NEEDED, I just thought it couldn't hurt anybody. I'll remove it and see if it still works

**Comment by MoonPadUSer** [comment]
I'll remove the variables completely and make one function that gives back the correctly formatted time

File: src/game/hud.cpp (hunk 5 above)
Comment (MoonPadUSer) [reply src/game/hud.cpp:42]: nope, they're not
